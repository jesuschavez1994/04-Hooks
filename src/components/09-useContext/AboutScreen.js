import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const AboutScreen = () => {

    const {User, setUser} = useContext(userContext);

    const handleReset = () => {
        setUser({});
    }

    return (
        <>
            <h1 className="ui header">
                AboutScreen
                <div className="sub header">Sub Header</div>
            </h1> 

            <pre>
                {JSON.stringify(User, null, 3)}
            </pre>

            <button className="ui inverted red button" onClick={handleReset}>Reset</button>
        </>
    )
}
