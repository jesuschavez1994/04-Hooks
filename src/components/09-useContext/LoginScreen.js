import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(userContext)

    return (
        <div>
            <h1 className="ui header">
                LoginScreen
                <div className="sub header">Sub Header</div>
            </h1> 

            <button className="ui inverted orange button" onClick={() => {
                setUser({
                    id:123,
                    name: 'jesus chavez'
                })
            }}>Orange</button>
        </div>
    )
}
