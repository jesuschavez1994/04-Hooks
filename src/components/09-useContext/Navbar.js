import React from 'react';
import { Link, NavLink } from "react-router-dom";

export const Navbar = () => {
    return (
        <div className="ui vertical pointing menu">
       
            <NavLink exact={true} activeClassName="active" to="/" className="item">Home</NavLink>

            <NavLink exact={true} activeClassName="active" to="/about" className="item">About</NavLink>
         
            <NavLink exact={true} activeClassName="active" to="/login" className="item">Login</NavLink>
       
        </div>
    )
}
