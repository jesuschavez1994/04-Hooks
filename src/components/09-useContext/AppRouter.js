import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { Navbar } from './Navbar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {

    return (
        <Router>
            <div>

                <Navbar />

                <Switch>
                    <Route exact={true} path="/" component={ HomeScreen }/>
                    <Route exact={true} path="/about" component={ AboutScreen }/>
                    <Route exact={true} path="/login" component={ LoginScreen }/>
                    <Route  component={ HomeScreen }/>
                </Switch>

            </div>
        </Router>
    )
}
