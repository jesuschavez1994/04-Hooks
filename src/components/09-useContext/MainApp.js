import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { userContext } from './UserContext'

export const MainApp = () => {

    const [User, setUser] = useState({})

    return (
        <userContext.Provider value={{
            User,
            setUser
        }}>
            <AppRouter />
        </userContext.Provider>
    )
}
