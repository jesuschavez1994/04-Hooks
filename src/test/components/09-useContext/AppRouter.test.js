
import { mount, shallow } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import '@testing-library/jest-dom';
import { userContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <AppRouter />', () => {

    const User = {
        id: 1,
        name: 'Jesus'
    }

    const wrapper = mount( 
        <userContext.Provider value ={{
            User
        }}>
            <AppRouter />
        </userContext.Provider>
    )

    test('debe de mostrar correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
