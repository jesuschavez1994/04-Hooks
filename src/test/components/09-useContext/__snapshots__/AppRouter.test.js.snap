// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar correctamente  1`] = `
<div>
  <div
    className="ui vertical pointing menu"
  >
    <a
      aria-current="page"
      className="item active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="item"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      aria-current={null}
      className="item"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  Array [
    <h1
      className="ui header"
    >
      HomeScreen
      <div
        className="sub header"
      >
        Sub Header
      </div>
    </h1>,
    <pre>
      {
   "id": 1,
   "name": "Jesus"
}
    </pre>,
  ]
</div>
`;
